core/src/main/java/org/pentaho/platform/scheduler2/action/DefaultActionInvoker.java

```diff
@@ -132,8 +133,12 @@ protected IActionInvokeStatus invokeActionImpl( final IAction actionBean,
     ActionUtil.removeKeyFromMap( params, ActionUtil.INVOKER_STREAMPROVIDER );
     ActionUtil.removeKeyFromMap( params, ActionUtil.INVOKER_UIPASSPARAM );
 
-    final ActionRunner
-      actionBeanRunner = new ActionRunner( actionBean, actionUser, params, streamProvider );
+    final IActionRunner actionBeanRunner = PentahoSystem.get( IActionRunner.class );
+    actionBeanRunner.setAction( actionBean );
+    actionBeanRunner.setActionUser( actionUser );
+    actionBeanRunner.setParams( params );
+    actionBeanRunner.setStreamProvider( streamProvider );
```

Comment:
As a possible _future_ improvement, I think it would be best to define a bean-configurable factory interface for the action runner, e.g. `IActionRunnerFactory#createActionRunner(actionBean, ...)`, allowing to keep the action runner class/interface immutable.